types/schemas_test.go

```diff
@@ -0,0 +1,76 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSchemas(t *testing.T) {
```

Comment:
Thanks for adding tests! However, this does not seem to cover the actual code that this PR is modifying, correct? Could you add a new case for that?

Reply:
It does, the code we added runs every time we run `AddSchema` on a schema which is an embedded type. In the test cases we embed the `hobbit` schema into the `baggins` schema which triggers the `doRemoveSchema` and removes the old `baggins` type while embedding.

Reply:
Oh I missed it since it was further down in the call stack, thanks for the clarification!

Reply:
All good, thanks for being thorough!